### PR TITLE
Add some VK_EXT_image_drm_format_modifier checks

### PR DESCRIPTION
**Pull request by spencer-lunarg**
part of https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/5632

adds

- VUID-VkImageCreateInfo-tiling-02353
- VUID-VkImageCreateInfo-pNext-06746
- VUID-vkGetImageDrmFormatModifierPropertiesEXT-image-02272
- VUID-VkImageSubresourceRange-aspectMask-02278
- VUID-VkPhysicalDeviceImageDrmFormatModifierInfoEXT-sharingMode-02314
- VUID-VkPhysicalDeviceImageDrmFormatModifierInfoEXT-sharingMode-02316

### CODE REVIEW

File: layers/stateless/sl_image.cpp [hunk 1]
```diff
@@ -367,24 +367,43 @@ bool StatelessValidation::manual_PreCallValidateCreateImage(VkDevice device, con
...
+    const auto format_list_info = LvlFindInChain<VkImageFormatListCreateInfo>(pCreateInfo->pNext);
+
     std::vector<uint64_t> image_create_drm_format_modifiers;
     if (IsExtEnabled(device_extensions.vk_ext_image_drm_format_modifier)) {
         const auto drm_format_mod_list = LvlFindInChain<VkImageDrmFormatModifierListCreateInfoEXT>(pCreateInfo->pNext);
         const auto drm_format_mod_explict = LvlFindInChain<VkImageDrmFormatModifierExplicitCreateInfoEXT>(pCreateInfo->pNext);
         if (pCreateInfo->tiling == VK_IMAGE_TILING_DRM_FORMAT_MODIFIER_EXT) {
-            if (((drm_format_mod_list != nullptr) && (drm_format_mod_explict != nullptr)) ||
-                ((drm_format_mod_list == nullptr) && (drm_format_mod_explict == nullptr))) {
+            if ((drm_format_mod_list != nullptr) && (drm_format_mod_explict != nullptr)) {
```
Comment (arno-lunarg) [inline layers/stateless/sl_image.cpp:377]: The condition in this `if` and the following `else if` seem to be inverted

File: layers/stateless/sl_image.cpp [hunk 2]
```diff
@@ -417,6 +436,12 @@ bool StatelessValidation::manual_PreCallValidateCreateImage(VkDevice device, con
                 }
             }
         }
+
+        const auto compression_control = LvlFindInChain<VkImageCompressionControlEXT>(pCreateInfo->pNext);
+        if (drm_format_mod_explict && compression_control) {
+            skip |= LogError("VUID-VkImageCreateInfo-pNext-06746", device, create_info_loc.dot(Field::pNext),
+                             "is has both VkImageCompressionControlEXT and VkImageDrmFormatModifierExplicitCreateInfoEXT.");
```
Comment (arno-lunarg) [inline layers/stateless/sl_image.cpp:443]:
```suggestion
                             "has both VkImageCompressionControlEXT and VkImageDrmFormatModifierExplicitCreateInfoEXT.");
```

File: layers/stateless/sl_instance_device.cpp [hunk 3]
```diff
@@ -738,11 +738,43 @@ bool StatelessValidation::manual_PreCallValidateGetPhysicalDeviceImageFormatProp
...
+                    skip |=
+                        LogError("VUID-VkPhysicalDeviceImageDrmFormatModifierInfoEXT-sharingMode-02315", physicalDevice,
                                  format_info_loc.pNext(Struct::VkPhysicalDeviceImageDrmFormatModifierInfoEXT, Field::sharingMode),
                                  "is VK_SHARING_MODE_CONCURRENT, but queueFamilyIndexCount is %" PRIu32 ".",
                                  image_drm_format->queueFamilyIndexCount);
+                } else if (!image_drm_format->pQueueFamilyIndices) {
+                    skip |= LogError(
+                        "VUID-VkPhysicalDeviceImageDrmFormatModifierInfoEXT-sharingMode-02314", physicalDevice,
+                        format_info_loc.pNext(Struct::VkPhysicalDeviceImageDrmFormatModifierInfoEXT, Field::sharingMode),
+                        "is VK_SHARING_MODE_CONCURRENT, queueFamilyIndexCount is %" PRIu32 ", but pQueueFamilyIndices is null.",
```
Comment (arno-lunarg) [inline layers/stateless/sl_instance_device.cpp:752]:
```suggestion
                        "is VK_SHARING_MODE_CONCURRENT, queueFamilyIndexCount is %" PRIu32 ", but pQueueFamilyIndices is NULL.",
```
For (I think) consistency with the rest of the code base

File: layers/stateless/sl_instance_device.cpp [hunk 4]
```diff
@@ -738,11 +738,43 @@ bool StatelessValidation::manual_PreCallValidateGetPhysicalDeviceImageFormatProp
...
+                        image_drm_format->queueFamilyIndexCount);
+                } else {
+                    uint32_t queue_family_property_count = 0;
+                    DispatchGetPhysicalDeviceQueueFamilyProperties2(physicalDevice, &queue_family_property_count, nullptr);
+                    vvl::unordered_set<uint32_t> queue_family_indices_set;
+                    for (uint32_t i = 0; i < image_drm_format->queueFamilyIndexCount; i++) {
+                        const uint32_t indices = image_drm_format->pQueueFamilyIndices[i];
```
Comment (arno-lunarg) [inline layers/stateless/sl_instance_device.cpp:759]:
```suggestion
                        const uint32_t queue_index = image_drm_format->pQueueFamilyIndices[i];
```

**Review by arno-lunarg — CHANGES_REQUESTED** [review]

File: layers/stateless/sl_image.cpp (hunk 1 above)
Comment (spencer-lunarg) [reply layers/stateless/sl_image.cpp:377]: good catch

Comment (arno-lunarg) [reply layers/stateless/sl_image.cpp]: I've got your back (⌐■_■)

**Review by arno-lunarg — APPROVED** [review]

**Pull request merged**